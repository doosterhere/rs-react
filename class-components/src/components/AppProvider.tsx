'use client';

import { ReactNode, useRef } from 'react';
import { Provider } from 'react-redux';

import '../styles/global.css';

import { AppStore, configStore } from '../store/store';
import { ThemeProvider } from '../components/ThemeContext';
import { ErrorBoundary } from './ErrorBoundary';

function AppProvider({ children }: { children: ReactNode }) {
  const storeRef = useRef<AppStore>();

  if (!storeRef.current) {
    storeRef.current = configStore();
  }

  return (
    <Provider store={storeRef.current}>
      <ErrorBoundary>
        <ThemeProvider>{children}</ThemeProvider>
      </ErrorBoundary>
    </Provider>
  );
}

export { AppProvider };
