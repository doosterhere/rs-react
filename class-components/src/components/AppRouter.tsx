import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import { ROUTES } from '../constants';

import { HomePage, Page404 } from '../pages';
import { ListItemDetailed } from './ListItemDetailed/ListItemDetailed';

const router = createBrowserRouter([
  {
    path: ROUTES.HOME,
    element: <HomePage />,
    errorElement: <Page404 />,
    children: [
      {
        path: ROUTES.DETAIL,
        element: <ListItemDetailed />,
      },
    ],
  },
]);

const AppRouter = () => <RouterProvider router={router} />;

export { AppRouter };
