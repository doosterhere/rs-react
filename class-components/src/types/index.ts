import { PlanetType } from './planet.type';
import { DefaultResponseType } from './default.response.type';

export type { PlanetType, DefaultResponseType };
