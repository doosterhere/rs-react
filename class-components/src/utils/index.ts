import { renderWithRouter } from './test-utils';

export { renderWithRouter };
