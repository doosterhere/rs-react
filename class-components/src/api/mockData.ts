import { DefaultResponseType, PlanetType } from '../types';

const mockDetailedData: PlanetType = {
  name: 'test-planet',
  diameter: '100',
  climate: 'test-climate',
  gravity: '200',
  terrain: 'test-terrain',
  surface_water: '300',
  population: '400',
  url: 'https://swapi.dev/api/planets/1/',
};

const mockData: DefaultResponseType<PlanetType> = {
  count: 2,
  results: [
    {
      name: 'test-planet-1',
      diameter: '100',
      climate: 'test-climate',
      gravity: '200',
      terrain: 'test-terrain',
      surface_water: '300',
      population: '400',
      url: 'https://swapi.dev/api/planets/1/',
    },
    {
      name: 'test-planet-2',
      diameter: '1100',
      climate: 'test-climate',
      gravity: '1200',
      terrain: 'test-terrain',
      surface_water: '1300',
      population: '1400',
      url: 'https://swapi.dev/api/planets/2/',
    },
  ],
};

export { mockDetailedData, mockData };
