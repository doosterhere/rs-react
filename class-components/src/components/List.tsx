import { Component } from 'react';
import { PlanetType } from '../types';

interface IListProps {
  itemsList: PlanetType[];
  isLoading: boolean;
}

class List extends Component<IListProps> {
  render() {
    return (
      <div className="list">
        {!!this.props.itemsList.length &&
          this.props.itemsList.map(item => (
            <div className="list-item" key={item.name}>
              {item.name}
            </div>
          ))}
        {!this.props.itemsList.length && !this.props.isLoading && (
          <div>No results found</div>
        )}
      </div>
    );
  }
}

export default List;
