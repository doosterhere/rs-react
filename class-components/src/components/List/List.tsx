import { FC } from 'react';

import classes from './List.module.css';

import { PlanetType } from '../../types';
import { ListItem } from '../ListItem';

interface IListProps {
  itemsList: PlanetType[] | [];
}

const List: FC<IListProps> = ({ itemsList }) => {
  return (
    <div className={classes.list} data-testid="list">
      {!!itemsList.length &&
        itemsList.map(item => {
          const id = item.url.match(/(\d+)/)?.[0] ?? '0';

          return <ListItem {...item} key={item.name} id={id} />;
        })}
    </div>
  );
};

export { List };
