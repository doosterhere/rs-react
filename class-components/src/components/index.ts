import SearchBar from './SearchBar.tsx';
import List from './List.tsx';
import ListItem from './ListItem.tsx';
import Loader from './Loader.tsx';
import ErrorBoundary from './ErrorBoundary.tsx';
import ButtonErrorTest from './ButtonErrorTest.tsx';
import Fallback from './Fallback.tsx';
import Modal from './Modal.tsx';
import AppRouter from './AppRouter.tsx';
import Layout from './Layout.tsx';

export {
  SearchBar,
  List,
  ListItem,
  Loader,
  ErrorBoundary,
  ButtonErrorTest,
  Fallback,
  Modal,
  AppRouter,
  Layout,
};
