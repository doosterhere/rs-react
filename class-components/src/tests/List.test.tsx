import { screen } from '@testing-library/react';

import { renderWithProvider } from '../utils';
import { List } from '../components/List';
import { mockedPlanets } from './testSetup/mockData';
import { PlanetType } from '../types';

jest.mock('../components/ListItem', () => ({
  ListItem: (item: PlanetType) => <div>{item.name}</div>,
}));

describe('List', () => {
  it('should renders correctly with data', () => {
    const { container } = renderWithProvider(<List itemsList={mockedPlanets.results} />);

    expect(container).toMatchSnapshot();

    expect(screen.getByText(/^Mocked\sPlanet$/i)).toBeInTheDocument();

    expect(screen.getByText(/^Mocked\sPlanet\s2$/i)).toBeInTheDocument();
  });

  it('should renders correctly without data', () => {
    const { container } = renderWithProvider(<List itemsList={[]} />);

    expect(container).toMatchSnapshot();

    expect(screen.getByTestId('list').childNodes).toHaveLength(0);
  });
});
