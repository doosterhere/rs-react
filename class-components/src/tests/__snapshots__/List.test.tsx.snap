// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should renders correctly with data 1`] = `
<div>
  <main
    class="main light"
  >
    <div
      class="list"
      data-testid="list"
    >
      <div>
        Mocked Planet
      </div>
      <div>
        Mocked Planet 2
      </div>
    </div>
  </main>
</div>
`;

exports[`List should renders correctly without data 1`] = `
<div>
  <main
    class="main light"
  >
    <div
      class="list"
      data-testid="list"
    />
  </main>
</div>
`;
